### ISSUE TITLE
Understanding TimeDistributed Layer in Keras

### ISSUE THREAD

**Issue by DeependraParichha1004** [keras-team/keras#16894]
**System information**.

TensorFlow version (2.9.1):
Are you willing to contribute it (No) :

**Describe the feature and the current behavior/state**.

Where does the use of TimeDistributed Layer come most? Is it significant to use this layer instead of the Dense layer, even after getting the same result?

**Comment by andreped** [comment]
@DeependraParichha1004 I would suggest asking these types of questions on stack overflow.

You are not supposed to swap TimeDistributed with a Dense layer (or similar). The function of the TimeDistributed layer is to wrap around another layer (or keras model) to apply a specific layer along the temporal axis, without storing replicas for each temporal item in memory ([see docs](https://keras.io/api/layers/recurrent_layers/time_distributed/) for more info).

Here is a nice tutorial which describes a common use case for TimeDistributed:
https://machinelearningmastery.com/timedistributed-layer-for-long-short-term-memory-networks-in-python/

In addition to sequence classification, I have been successful in using it for multiple instance learning (MIL) (see [this](https://keras.io/examples/vision/attention_mil_classification/) example if MIL is unfamiliar to you).

Since you said that you got the same results, what was your use case? If you are training a simple image classifier, then TimeDistributed is not relevant. It depends on the context.

Here is an example of using TimeDistributed (use case could be video classification):
```
import keras
from keras.applications import MobileNetV2

input_ = keras.layers.Input(shape=(8, 224, 224, 3))
base_model  = MobileNetV2(include_top=True, input_shape=(224, 224, 3))
output = keras.layers.TimeDistributed(base_model)(input_)
model = keras.Model(inputs=input_, outputs=output)
```

**Comment by tilakrayal** [comment]
@DeependraParichha1004,
Could you please take a look at this [comment](https://github.com/keras-team/keras/issues/16894#issuecomment-1214919423) and also the doc [link](https://keras.io/api/layers/recurrent_layers/time_distributed/) for the required information reg. Time Distributed Layer and let us know if you are looking for the same. 

In `keras` - while building a sequential model - usually the second dimension (one after sample dimension) - is related to a time dimension. This means that if for example, your data is 5-dim with (sample, time, width, length, channel) you could apply a `convolutional` layer using `TimeDistributed` (which is applicable to 4-dim with (sample, width, length, channel)) along a time dimension (applying the same layer to each time slice) in order to obtain 5-d output.

The case with Dense is that in keras from version 2.0 Dense is by default applied to only last dimension (e.g. if you apply Dense(10) to input with shape (n, m, o, p) you'll get output with shape (n, m, o, 10)) so in your case Dense and `TimeDistributed`(Dense) are equivalent. Thank you!

**Comment by DeependraParichha1004** [comment]
Thank you @andreped @tilakrayal for your valuable time. I think I misunderstood the concept.

**Comment by tilakrayal** [comment]
@DeependraParichha1004,
Glad the issue is resolved for you, Could you please feel free to move this to closed status. Thank you!

**Comment by andreped** [comment]
@tilakrayal As he has not replied after 5 days, I would just close this issue for him, as it has been solved.

**Comment by DeependraParichha1004** [comment]
Thankyou @andreped @tilakrayal and sorry for the inconvenience.

**Comment by DeependraParichha1004** [comment]
As

**Comment by andreped** [comment]
@DeependraParichha1004 No worries! Hope you find TimeDistributed more beneficial in future projects :]
